tests/test_mm_packing_robustness.py

```
@@ -0,0 +1,163 @@
+
+class TestTruncateLongSeqMultimodal:
+    """truncate_long_seq guards multimodal rows."""
+
+    def test_multimodal_row_over_length_raises(self):
+        sample = {
+            "input_ids": [list(range(10))],
+            "labels": [list(range(10))],
+            "pixel_values": [[[0.0, 1.0], [2.0, 3.0]]],
+            "length": [10],
+        }
+        with pytest.raises(ValueError, match="(?i)multimodal"):
```

Ruff (0.15.20)

[warning] Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)